### PR TITLE
Add jMetrik 4.1.1

### PR DESCRIPTION
**Pull request by sbibauw**
jMetrik is a free psychometric measurement and statistics software
distributed by Psychomeasurement Systems, LLC.
It runs on Java (set as dependency) and is being installed by
install4j. I heavily copied the formula for SoapUI.rb and adapted
the parameters for the install4j install and uninstall scripts.

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/jmetrik.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+cask 'jmetrik' do
+  version '4.1.1'
+  sha256 'a6b7fa7870232f9bf615704c810c8046b3b5ebc02ec3a920fb96e0f255b61321'
+
+  url "https://itemanalysis.com/jmetrik/v#{version.dots_to_underscores}/jmetrik_macos_#{version.dots_to_underscores}_java7.dmg"
+  name 'jMetrik'
+  homepage 'http://itemanalysis.com/'
+
+  depends_on cask: 'java'
```
Comment (vitorgalvao) [inline Casks/jmetrik.rb]: Needs to be a `caveat` instead.

https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md#caveats-mini-dsl

File: Casks/jmetrik.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  # installer :manual => "jMetrik Installer.app"
```
Comment (vitorgalvao) [inline Casks/jmetrik.rb]: Remove this.

File: Casks/jmetrik.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  # Installer runs install4j from the distribution in quiet mode.
+  #
+  # This Installer is copied and adapted from the formula for SoapUI.
```
Comment (vitorgalvao) [inline Casks/jmetrik.rb]: Don’t mention where it’s copied from. If that one is changed, this will be outdated.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/jmetrik.rb (hunk 1 above)
Comment (sbibauw) [reply Casks/jmetrik.rb]: OK, thank! All done!

File: Casks/jmetrik.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  #
+  installer script: {
+                      executable: 'jMetrik Installer.app/Contents/MacOS/JavaApplicationStub',
+                      args:       [
+                                    '-q',
+                                    '-VcreateDesktopLinkAction$Boolean=false',
+                                    '-VexecutionLauncherAction$Boolean=false',
+                                    '-Vsys.adminRights$Boolean=true',
+                                    '-Vsys.installationDir=/Applications/jMetrik',
+                                    '-Vsys.languageId=en',
```
Comment (reitermarkus) [inline Casks/jmetrik.rb]: I don't think this is needed.

File: Casks/jmetrik.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  # Refer to the install4j command line options for additional information
+  #   https://resources.ej-technologies.com/install4j/help/doc/index.html
+  #
+  installer script: {
+                      executable: 'jMetrik Installer.app/Contents/MacOS/JavaApplicationStub',
+                      args:       [
+                                    '-q',
+                                    '-VcreateDesktopLinkAction$Boolean=false',
+                                    '-VexecutionLauncherAction$Boolean=false',
+                                    '-Vsys.adminRights$Boolean=true',
```
Comment (reitermarkus) [inline Casks/jmetrik.rb]: I don't think this is needed.

File: Casks/jmetrik.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                                    '-Vsys.installationDir=/Applications/jMetrik',
+                                    '-Vsys.languageId=en',
+                                    '-Vsys.programGroupDisabled$Boolean=true',
+                                  ],
+                      sudo:       true,
```
Comment (reitermarkus) [inline Casks/jmetrik.rb]: Does this actually require `sudo`?

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: Casks/jmetrik.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  #   https://resources.ej-technologies.com/install4j/help/doc/index.html
+  #
+  installer script: {
+                      executable: 'jMetrik Installer.app/Contents/MacOS/JavaApplicationStub',
+                      args:       [
+                                    '-q',
+                                    '-VcreateDesktopLinkAction$Boolean=false',
+                                    '-VexecutionLauncherAction$Boolean=false',
+                                    '-Vsys.adminRights$Boolean=true',
+                                    '-Vsys.installationDir=/Applications/jMetrik',
```
[the remark this answers is not in the document]
Comment (sbibauw) [reply Casks/jmetrik.rb]: OK. Done.

File: Casks/jmetrik.rb (hunk 4 above)
Comment (sbibauw) [reply Casks/jmetrik.rb]: Indeed. It seems to be the default.

File: Casks/jmetrik.rb (hunk 5 above)
Comment (sbibauw) [reply Casks/jmetrik.rb]: Indeed. It seems to be the default.

File: Casks/jmetrik.rb (hunk 6 above)
Comment (sbibauw) [reply Casks/jmetrik.rb]: Yes! The install4j Installer does require admin rights to both install and uninstall the app. If I do not mention "sudo: true" here, the Terminal will request admin rights via an external window during the installation.

**Comment by vitorgalvao** [comment]
This looks good. Would you consider doing the same to the SoapUI cask?

**Comment by sbibauw** [comment]
> This looks good. Would you consider doing the same to the SoapUI cask?

Done! (PR #58706)

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**